### PR TITLE
chore(withdraws): standalone pages for withdraw and transfer

### PR DESCRIPTION
**Pull request by macqbat**
# Related issues 🔗

Issue: #4646 & #4647 
# Description ℹ️

Dedicated pages fot transfer and withdrawal tickets. Contains handling of assetId pass and clear it for not breaking the params state

# Demo 📺

image
image

# Technical 👨‍🔧

Details of technical implementation that reviewers may need to be aware of, if applicable.

### CODE REVIEW

File: apps/trading/client-pages/transfer/transfer.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  TransferContainer,
+  useTransferAssetIdStore,
+} from '@vegaprotocol/accounts';
+import { useSearchParams } from 'react-router-dom';
+import { useEffect } from 'react';
+
+export const Transfer = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const assetId = searchParams.get('assetId') || undefined;
+  const selectedAssetId = useTransferAssetIdStore((store) => store.assetId);
```
Comment (mattrussell36) [inline apps/trading/client-pages/transfer/transfer.tsx:13]: I don't see any purpose for the useTransferAssetIdStore.

File: apps/trading/client-pages/withdraw/withdraw.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { WithdrawContainer } from '../../components/withdraw-container';
+import { useEffect } from 'react';
+
+export const Withdraw = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const assetId = searchParams.get('assetId') || undefined;
+  const asset = useWithdrawStore((store) => store.asset);
+  useEffect(() => {
+    if (assetId && asset?.id && assetId !== asset.id) {
+      setSearchParams({});
```
Comment (mattrussell36) [inline apps/trading/client-pages/withdraw/withdraw.tsx:14]: I don't think we need to worry about clearing the url if the asset changes.

File: apps/trading/pages/client-router.tsx [hunk 3]
```diff
@@ -35,6 +35,14 @@ const LazyDeposit = dynamic(() => import('../client-pages/deposit'), {
   ssr: false,
 });
 
+const LazyWithdraw = dynamic(() => import('../client-pages/withdraw'), {
```
Comment (mattrussell36) [inline apps/trading/pages/client-router.tsx:38]: I'd like to move away from using next/dynamic and even React.lazy in favor of just having a single js bundle. This makes things easier with ipfs and the benefits we gain from code splitting are minimal due to the nature of this being more app-like

File: apps/trading/pages/client-router.tsx [hunk 4]
```diff
@@ -104,6 +116,8 @@ const routerConfig: RouteObject[] = [
     element: <LazyDisclaimer />,
   },
   { path: Routes.DEPOSIT, element: <LazyDeposit /> },
+  { path: Routes.WITHDRAW, element: <LazyWithdraw /> },
```
Comment (mattrussell36) [inline apps/trading/pages/client-router.tsx:119]: (non blocking): We should probably wrap these with a layout route to ensure consistency

File: libs/accounts/src/lib/transfer-form.tsx [hunk 5]
```diff
@@ -392,3 +396,16 @@ export const AddressField = ({
     </>
   );
 };
+
+interface TransferAssetIdStore {
+  assetId?: string;
+  setAssetId: (assetId: string) => void;
+}
+export const useTransferAssetIdStore = create<TransferAssetIdStore>()(
```
Comment (mattrussell36) [inline libs/accounts/src/lib/transfer-form.tsx:404]: I think this store is unneccessary

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading/client-pages/transfer/transfer.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+jest.mock('@vegaprotocol/accounts', () => ({
+  ...jest.requireActual('@vegaprotocol/accounts'),
+  TransferContainer: jest.fn(({ assetId }: { assetId?: string }) => (
+    <div data-testid="assetId">{assetId}</div>
+  )),
+}));
+
+const renderJsx = () => {
+  render(
+    <VegaWalletContext.Provider value={mockWalletContext}>
```
Comment (mattrussell36) [inline apps/trading/client-pages/transfer/transfer.spec.tsx:29]: This is not needed as the container is mocked

Comment (mattrussell36) [reply apps/trading/client-pages/transfer/transfer.spec.tsx:29]: Actually it was needed for the GetStarted component, but I think it simpler to mock it. I've pushed changes

File: apps/trading/pages/client-router.tsx (hunk 4 above)
Comment (mattrussell36) [reply apps/trading/pages/client-router.tsx]: I'll make an additional PR for this

File: libs/accounts/src/lib/transfer-form.tsx (hunk 5 above)
Comment (mattrussell36) [reply libs/accounts/src/lib/transfer-form.tsx]: Having a whole new store to clear the query string is overkill.

File: apps/trading/pages/client-router.tsx (hunk 3 above)
Comment (mattrussell36) [reply apps/trading/pages/client-router.tsx]: I've changed these to just be imported in a regular fashion. I'll make an additional PR to convert the rest

**Review by mattrussell36 — APPROVED** [review]

**Comment by bwallacee** [comment]
@macqbat On the withdraws page, if I disconnect my etherum wallet and then select connect again, It connects straight away with no messages or popups etc on my metamask wallet, is this expected?

**Comment by bwallacee** [comment]
If I try to scroll down on the withdraw assets list, it flashes and brings me back up to the top. Meaning I am unable to select any assets.
https://github.com/vegaprotocol/frontend-monorepo/assets/133853298/ae89ece1-099c-4c2e-a16b-2092bddb2fa2

**Comment by bwallacee** [comment]
I have raised a new issue for a withdraw scroll as it's happening on the deal ticket also:
https://github.com/vegaprotocol/frontend-monorepo/issues/4741

**Comment by bwallacee** [comment]
Should we add new specs within this PR?

**Comment by mattrussell36** [comment]
> @macqbat On the withdraws page, if I disconnect my etherum wallet and then select connect again, It connects straight away with no messages or popups etc on my metamask wallet, is this expected?

@bwallacee this is expected. Because the wallet already allowed connection to this website it doesn't require you to confirm the connection a second time

**Comment by mattrussell36** [comment]
@macqbat @bwallacee 

I've added a single spec for each page to capture the setting the asset via the assetId query param. I don't think we need additional specs as everything else works exactly the same as when the component is opened in the sidebar.

**Pull request merged**